types/vfile/index.d.ts

```diff
@@ -0,0 +1,85 @@
+// Type definitions for vfile 2.2
+// Project: https://github.com/vfile/vfile#readme
+// Definitions by: Junyoung Choi <https://github.com/rokt33r>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare namespace vfile {
+    type Contents = string | Buffer;
+
+    interface Point {
+        line: number;
+        column: number;
+    }
+
+    interface Position {
+        start: Point;
+        end: Point;
+    }
+
+    interface Node {
+        type: string;
+        data?: {};
+        position?: Position;
+    }
+
+    interface VFileParamsBase<D> {
+        data?: D;
+        contents?: Contents;
+        path?: string;
+        basename?: string;
+        stem?: string;
+        extname?: string;
+        dirname?: string;
+        cwd?: string;
+    }
+
+    type VFileParams<C, D> = VFileParamsBase<D> & C;
+
+    interface VFileError {
+        ruleId: string | null;
+        name: string;
+        file: string;
+        reason: string;
+        line: number | null;
+        column: number | null;
+        location: Position;
+        source: string | null;
+        fatal?: boolean | null;
+    }
+
+    type StringifiablePosition = Point | Position | Node;
+
+    type Message = (reason: string, position?: StringifiablePosition, ruleId?: string) => VFileError;
+
+    type Fail = (reason?: string, position?: StringifiablePosition, ruleId?: string) => void;
```

Comment:
`fail` throws the error btw!

Reply:
As long as I know, we cannot define function types that always throw an error.

But, it should be better to use `never` for return type rather than `void`.
https://basarat.gitbooks.io/typescript/docs/types/never.html

Reply:
Yes!

```
A function always throws (e.g. in function `foo(){throw new Error('Not Implemented')}` the return type of `foo` is never)
```